pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -333,37 +333,59 @@ func (c *Controller) syncNetworkWithGNP(ctx context.Context, network *networkv1.
 	return nil
 }
 
-func (c *Controller) handleGKENetworkParamSetDelete(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
+// handleGNPDelete checks to see if its safe to delete the GNP resource before calling executeGNPDelete on it
+func (c *Controller) handleGNPDelete(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
 	if params.Status.NetworkName == "" {
-		removeFinalizerInPlace(params)
-		return nil
+		return c.executeGNPDelete(ctx, params, nil)
 	}
 
 	network, err := c.networkClientset.NetworkingV1().Networks().Get(ctx, params.Status.NetworkName, v1.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
-			removeFinalizerInPlace(params)
-			return nil
+			return c.executeGNPDelete(ctx, params, nil)
 		}
 		return err
 	}
 
 	networkStillRefersToGNP := network.Spec.ParametersRef != nil && strings.EqualFold(network.Spec.ParametersRef.Kind, gnpKind) && network.Spec.ParametersRef.Name == params.Name
 	if !networkStillRefersToGNP {
-		removeFinalizerInPlace(params)
-		return nil
+		return c.executeGNPDelete(ctx, params, network)
 	}
 
 	if networkStillRefersToGNP && !network.InUse() {
-		removeFinalizerInPlace(params)
-		return nil
+		return c.executeGNPDelete(ctx, params, network)
 	}
 
 	// if the network is in use, this GNP object will get reconciled again when the network's in use status changes.
 
 	return nil
 }
 
+func (c *Controller) executeGNPDelete(ctx context.Context, params *networkv1.GKENetworkParamSet, network *networkv1.Network) error {
+	removeFinalizerInPlace(params)
+
+	if network != nil {
+		newNetwork := network.DeepCopy()
+		conditionRemoved := false
+		for i, f := range newNetwork.Status.Conditions {
+			if f.Type == string(networkv1.NetworkConditionStatusParamsReady) {
+				newNetwork.Status.Conditions = append(newNetwork.Status.Conditions[:i], newNetwork.Status.Conditions[i+1:]...)
```

Comment:
Do not delete the whole condition, just set it false.

Reply:
waiting for #562 so I can add a reason